app/javascript/controllers/partner_form_validation_controller.js

```diff
@@ -0,0 +1,59 @@
+import { Controller } from "@hotwired/stimulus";
+import _ from "lodash";
+
+// PartnerFormValidationController
+export default class extends Controller {
+	static targets = ["source"];
+
+	connect() {
+		const inputDebouncePeriod = 200;
+		const baseUrl = "/partners/lookup_name";
+
+		const csrfToken = document
+			.querySelector('meta[name="csrf-token"]')
+			.getAttribute("content");
+
+		const endpointHeaders = {
+			method: "GET",
+			headers: {
+				Accept: "application/json",
+				"Content-Type": "application/json",
+				"X-CSRF-Token": csrfToken,
+			},
+		};
+
+		const nameProblemFeedbackElement = document.getElementById(
+			"partner-name-feedback"
+		);
+
+		this.inputFunction = _.debounce(() => {
+			this.sourceTarget.classList.remove("is-invalid");
+			nameProblemFeedbackElement.style.display = "none";
+
+			const nameValue = this.sourceTarget.value;
+			if (nameValue.length < 1) {
```

Comment:
we don't allow names less than 5 characters so they can't collide till you hit that amount

Reply:
Fixed.